content/blog/2022/2022-08-30-handling-timezones-in-spring.md

```diff
@@ -0,0 +1,516 @@
+
+---
+title: "Handling timezones in a Spring Boot Application"
+categories: ["Spring"]
+date: 2022-08-30 00:00:00 +1100
+modified: 2022-08-30 00:00:00 +1100
+authors: ["ranjani"]
+description: "Handling timezones in a Spring Boot Application"
+image: images/stock/0111-clock-1200x628-branded.jpg
+url: spring-timezones
+---
+
+It is common to encounter applications that run on different timezones. Handling date and time operations across multiple layers of an application can be tricky.
+In this article, we will try to understand the options available in Java and map it in the context of a Spring application to effectively handle timezones.
+
+{{% github "https://github.com/thombergs/code-examples/tree/master/spring-boot/timezones" %}}
+
+## Understanding GMT, UTC and DST
+- **Greenwich Mean Time (GMT)** is a timezone used in some parts of the world, mainly Europe and Africa.
+GMT was replaced as the international standard time in 1972 by UTC.
+- **Universal Coordinated Time (UTC)** is not a timezone. It is a universally preferred standard that can be used to display timezones.
+- **Daylight Savings Time (DST)** is the practice of setting clocks forward by one hour in the summer months
+and back again in the fall, to make better use of natural daylight. **Neither GMT or UTC get affected by DST**. To account for DST changes,
+countries or states usually switch to another timezone. For instance in Australian summer, the states that observe DST will move from
+Australian Eastern Standard Time (AEST) to Australian Eastern Daylight Time (AEDT).
+
+Operations around dates, time and timezones can be confusing and prone to errors. To understand some problems around dates refer to this [article.](https://yourcalendricalfallacyis.com/)
+We will deep-dive into various aspects of handling timezones in the further sections.
+
+## Drawbacks of legacy time-based java.util classes
+
+Let's look at a few reasons why you should choose to avoid the date-time classes in the java.util package when developing applications.
+
+### Missing Timezone information
+
+{{% image alt="settings" src="images/posts/handling-timezones-in-spring/date.JPG" %}}
+
+- `java.util.Date` represents an instant in time. 
+- Also, it has no timezone information. So, it considers a **default system timezone** which could differ from location to location.
+For instance, if a person runs this test from another country, he might see a different date and time derived from the given milliseconds.
+
+### Custom Date creation
+{{% image alt="settings" src="images/posts/handling-timezones-in-spring/customDate.JPG" %}}
```

Comment:
Instead of a screenshot of code and console output, can you add the code and the console output into the article itself? That's better readable and more inclusive.